benchmarks/benchmarks/bench_lib.py

```diff
@@ -54,3 +54,64 @@ def setup(self, shape, pad_width, mode):
 
     def time_pad(self, shape, pad_width, mode):
         np.pad(self.array, pad_width, mode)
+
+class Nan(Benchmark):
+    """Benchmarks for nan functions"""
+
+    param_names = ["array_size", "percent_nans"]
+    params = [
+            # sizes of the 1D arrays
+            [200, int(2e5)],
+            # percent of np.nan in arrays
+            [0, 0.1, 2., 50., 90.],
+            ]
+
+    def setup(self, array_size, percent_nans):
+        np.random.seed(123)
+        # produce a randomly shuffled array with the
+        # approximate desired percentage np.nan content
+        base_array = np.random.randn(array_size)
```

Comment:
This has to be `rand`, otherwise the selection doesn't give the right percentiles! Maybe just use `np.random.uniform(size=array_size)` for explicitness.

Reply:
ok, done